src/uct/ib/mlx5/gdaki/gdaki.cuh

```diff
@@ -0,0 +1,354 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2025. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCT_GDAKI_CUH_H
+#define UCT_GDAKI_CUH_H
+
+#include <stdio.h>
+#include <stdint.h>
+
+#include <memory>
+#include <cooperative_groups.h>
+
+#include <doca_gpunetio_verbs_def.h>
+#include <doca_gpunetio_dev_verbs_qp.cuh>
+
+#include <ucs/type/status.h>
+#include <uct/api/cuda/uct.h>
+
+#include "gdaki_dev.h"
+
+#define UCT_RC_MLX5_GDA_WARP_SIZE 32
+
+template<uct_device_level_t level>
+UCS_F_DEVICE void
+uct_rc_mlx5_gda_exec_init(unsigned *lane_id, unsigned *num_lanes)
+{
+    switch (level) {
+    case UCT_DEVICE_LEVEL_THREAD:
+        *lane_id   = 0;
+        *num_lanes = 1;
+        break;
+    case UCT_DEVICE_LEVEL_WARP:
+        *lane_id   = doca_gpu_dev_verbs_get_lane_id();
+        *num_lanes = UCT_RC_MLX5_GDA_WARP_SIZE;
+        break;
+    case UCT_DEVICE_LEVEL_BLOCK:
+        *lane_id   = threadIdx.x;
+        *num_lanes = blockDim.x;
+        break;
+    case UCT_DEVICE_LEVEL_GRID:
+        *lane_id   = threadIdx.x + blockIdx.x * blockDim.x;
+        *num_lanes = blockDim.x * gridDim.x;
+        break;
+    }
+}
+
+template<uct_device_level_t level> UCS_F_DEVICE void uct_rc_mlx5_gda_sync(void)
+{
+    switch (level) {
+    case UCT_DEVICE_LEVEL_WARP:
+        __syncwarp();
+        break;
+    case UCT_DEVICE_LEVEL_BLOCK:
+        __syncthreads();
+        break;
+    case UCT_DEVICE_LEVEL_THREAD:
+        break;
+    case UCT_DEVICE_LEVEL_GRID:
+        auto g = cooperative_groups::this_grid();
+        g.sync();
+    }
+}
+
+UCS_F_DEVICE uint64_t uct_rc_mlx5_gda_reserv_wqe_thread(
+        struct doca_gpu_dev_verbs_qp *qp, unsigned count)
```

Comment:
`struct` is redundant in this file. Please remove here and for other type names.
```suggestion
        doca_gpu_dev_verbs_qp &qp, unsigned count)
```